servicetalk-http-api/src/main/java/io/servicetalk/http/api/ContentCoding.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright © 2020 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.http.api;
+
+/**
+ * API for HTTP <a href="https://tools.ietf.org/html/rfc7231#section-3.1.2.1">Content Codings</a>.
+ */
+public interface ContentCoding {
+
+    /**
+     * A string representation for the content coding.
+     *
+     * @return a string representation for the content coding.
+     */
+    String name();
```

Comment:
This API is used only by ST, consider using `CharSequence` here to avoid internal conversions `toString()`. We have `CharSequences` utility class that does case-insensitive comparisons.